### PR TITLE
Further MergeTree Client Legacy Deprecations

### PR DESCRIPTION
**Pull request by anthony-murphy**
In an effort the reduce exposure of the Client class in the merge-tree package this change additionally deprecates a number of types which either directly or indirectly expose the merge-tree Client class.

Most of these types are not meant to be used directly, and direct use is not supported:
 - AttributionPolicy
 - IClientEvents
 - IMergeTreeAttributionOptions
 - SharedSegmentSequence
 - SharedStringClass

Some of the deprecations are for class constructors and in those cases we plan to replace the class with an interface which has an equivalent API. Direct instantiation of these classes is not currently supported or necessary for any supported scenario, so the change to an interface should not impact usage:
- SequenceInterval
- SequenceEvent
- SequenceDeltaEvent
- SequenceMaintenanceEvent

### CODE REVIEW

File: .changeset/shaggy-pens-occur.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+"@fluidframework/merge-tree": minor
+"@fluidframework/sequence": minor
+---
+---
+"section": deprecation
+---
+
+Further MergeTree Client Legacy Deprecations
+
+In an effort the reduce exposure of the Client class in the merge-tree package this change additionally deprecates a number of types which either directly or indirectly expose the merge-tree Client class.
```
Comment (jzaffiro) [inline .changeset/shaggy-pens-occur.md:11]:
```suggestion
In an effort the reduce exposure of the Client class in the merge-tree package, this change additionally deprecates a number of types which either directly or indirectly expose the merge-tree Client class.
```

Comment (jzaffiro) [inline .changeset/shaggy-pens-occur.md:11]:
```suggestion
In an effort to reduce the exposure of the Client class in the merge-tree package this change additionally deprecates a number of types which either directly or indirectly expose the merge-tree Client class.
```

File: .changeset/shaggy-pens-occur.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+Most of these types are not meant to be used directly, and direct use is not supported:
+ - AttributionPolicy
+ - IClientEvents
+ - IMergeTreeAttributionOptions
+ - SharedSegmentSequence
+ - SharedStringClass
+
+Some of the deprecations are for class constructors and in those cases we plan to replace the class with an interface which has an equivalent API. Direct instantiation of these classes is not currently supported or necessary for any supported scenario, so the change to an interface should not impact usage:
```
Comment (jzaffiro) [inline .changeset/shaggy-pens-occur.md:20]:
```suggestion
Some of the deprecations are for class constructors, and in those cases, we plan to replace the class with an interface which has an equivalent API. Direct instantiation of these classes is not currently supported or necessary for any supported scenario, so the change to an interface should not impact usage:
```

File: packages/dds/sequence/src/intervals/sequenceInterval.ts [hunk 3]
```diff
@@ -127,6 +127,10 @@ export class SequenceInterval implements ISerializableInterval {
 		);
 	}
 
+	/**
+	 *
```
Comment (jzaffiro) [inline packages/dds/sequence/src/intervals/sequenceInterval.ts:131]:
```suggestion
```

File: packages/dds/sequence/src/sequenceDeltaEvent.ts [hunk 4]
```diff
@@ -127,6 +130,10 @@ export class SequenceDeltaEvent extends SequenceEvent<MergeTreeDeltaOperationTyp
 	 */
 	public readonly isLocal: boolean;
 
+	/**
+	 *
```
Comment (jzaffiro) [inline packages/dds/sequence/src/sequenceDeltaEvent.ts:134]:
```suggestion
```

File: packages/dds/sequence/src/sequenceDeltaEvent.ts [hunk 5]
```diff
@@ -148,6 +155,10 @@ export class SequenceDeltaEvent extends SequenceEvent<MergeTreeDeltaOperationTyp
  * @alpha
  */
 export class SequenceMaintenanceEvent extends SequenceEvent<MergeTreeMaintenanceType> {
+	/**
+	 *
```
Comment (jzaffiro) [inline packages/dds/sequence/src/sequenceDeltaEvent.ts:159]:
```suggestion
```

File: .changeset/shaggy-pens-occur.md (hunk 1 above)
Comment (tylerbutler) [inline .changeset/shaggy-pens-occur.md:11]:
```suggestion
To reduce exposure of the `Client` class in the merge-tree package, several types have been deprecated. These types directly or indirectly expose the merge-tree `Client` class.
```

File: .changeset/shaggy-pens-occur.md [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+---
+"section": deprecation
+---
+
+Further MergeTree Client Legacy Deprecations
+
+In an effort the reduce exposure of the Client class in the merge-tree package this change additionally deprecates a number of types which either directly or indirectly expose the merge-tree Client class.
+
+Most of these types are not meant to be used directly, and direct use is not supported:
+ - AttributionPolicy
```
Comment (tylerbutler) [inline .changeset/shaggy-pens-occur.md:14]:
```suggestion
Most of these types are not meant to be used directly, and direct use is not supported:

 - AttributionPolicy
```

File: .changeset/shaggy-pens-occur.md (hunk 2 above)
Comment (tylerbutler) [inline .changeset/shaggy-pens-occur.md:20]:
```suggestion
Some of the deprecations are class constructors. In those cases, we plan to replace the class with an interface which has an equivalent API. Direct instantiation of these classes is not currently supported or necessary for any supported scenario, so the change to an interface should not impact usage. This applies to the following types:
```

**Review by tylerbutler — APPROVED** [review]
Changeset looks great - just wording tweaks and minor formatting. Your level of detail was great.

Typically I would suggest trying to link to the docs for the types, but in this case that seems unnecessary since these types aren't likely to be used right now anyway. Maybe links to docs for deprecations in general is unnecessary. Links to docs make way more sense with new or changed APIs. @microsoft/fluid-cr-docs Curious what others think. We can take that discussion to another forum though.

File: .changeset/shaggy-pens-occur.md [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+"@fluidframework/merge-tree": minor
+"@fluidframework/sequence": minor
+---
+---
+"section": deprecation
+---
+
+Further MergeTree Client Legacy Deprecations
+
+In an effort the reduce exposure of the Client class in the merge-tree package, this change additionally deprecates a number of types which either directly or indirectly expose the merge-tree Client class.
```
Comment (anthony-murphy) [inline .changeset/shaggy-pens-occur.md:11]:
```suggestion
In an effort to reduce the exposure of the `Client` class in the merge-tree package, this change additionally deprecates a number of types which either directly or indirectly expose the merge-tree `Client` class.
```

File: .changeset/shaggy-pens-occur.md [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+"@fluidframework/merge-tree": minor
+"@fluidframework/sequence": minor
+---
+---
+"section": deprecation
+---
+
+Further MergeTree Client Legacy Deprecations
+
+In an effort to reduce the exposure of the `Client` class in the merge-tree package, this change additionally deprecates a number of types which either directly or indirectly expose the merge-tree `Client` class.
```
Comment (anthony-murphy) [inline .changeset/shaggy-pens-occur.md:11]:
```suggestion
To reduce exposure of the `Client` class in the merge-tree package, several types have been deprecated. These types directly or indirectly expose the merge-tree `Client` class.
```

File: .changeset/shaggy-pens-occur.md [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+
+Most of these types are not meant to be used directly, and direct use is not supported:
+
+ - AttributionPolicy
+ - IClientEvents
+ - IMergeTreeAttributionOptions
+ - SharedSegmentSequence
+ - SharedStringClass
+
+Some of the deprecations are for class constructors, and in those cases, we plan to replace the class with an interface which has an equivalent API. Direct instantiation of these classes is not currently supported or necessary for any supported scenario, so the change to an interface should not impact usage:
```
Comment (anthony-murphy) [inline .changeset/shaggy-pens-occur.md:21]:
```suggestion
Some of the deprecations are class constructors. In those cases, we plan to replace the class with an interface which has an equivalent API. Direct instantiation of these classes is not currently supported or necessary for any supported scenario, so the change to an interface should not impact usage. This applies to the following types:
```

**Review by Abe27342 — APPROVED** [review]

**Pull request merged**